smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -82,7 +82,14 @@ public KafkaSource(Vertx vertx, Config config, String servers) {
                             // The Kafka subscription must happen on the subscription.
                             this.consumer.subscribe(topic);
                         }))
-                .map(rec -> new IncomingKafkaRecord<>(consumer, rec));
+                .map(rec -> {
+                        // TODO: the actual commit strategy should be implemented
+                        return Message.newBuilder()
+                            .payload(rec.value())
+                            .metadata(new IncomingKafkaRecordMetadata<>(rec))
```

Comment:
If we move to builders for `Message`, should we also create builders for `Metadata`? Or is that specific to connectors so ok?

Reply:
This would make sense when there is lot of parameters involved. I didn't want to refactor the metadata in this PR. So the current status is that `OutgoingKafkaRecordMetadata` has a builder but not `IncomingKafkaRecordMetadata`. Maybe it is because the end user is more likely to create an `OutgoingKafkaRecordMetadata` (?).